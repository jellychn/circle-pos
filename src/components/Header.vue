<template>
  <div class="header">
    <div class="container">
      <IconCirclePos class="logo" @click="navigateToHome()" />
    </div>
  </div>
</template>

<script setup lang="ts">
import { useRouter } from 'vue-router'
import IconCirclePos from './icons/IconCirclePos.vue'

const router = useRouter()

function navigateToHome(): void {
  router.push('/')
}
</script>

<style lang="scss" scoped>
.header {
  z-index: 10;
  width: 100%;
  padding: 20px;
  position: fixed;
  top: 0;
  left: 0;
  background-color: var(--vt-c-white);

  .container {
    display: flex;
    max-width: var(--vt-max-width);
    margin: auto;

    .logo {
      cursor: pointer;
    }
  }
}
</style>