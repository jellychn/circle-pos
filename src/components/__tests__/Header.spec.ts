import { shallowMount } from '@vue/test-utils'
import { describe, it, expect, vi } from 'vitest'
import { useRouter } from 'vue-router'
import Header from '@/components/Header.vue'

vi.mock('vue-router')
vi.mocked(useRouter).mockReturnValue({
  push: vi.fn()
})

describe('Header.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('calls navigateToHome when logo is clicked', async () => {
    const wrapper = shallowMount(Header)

    await wrapper.find('.logo').trigger('click')

    expect(useRouter().push).toHaveBeenCalledWith('/')
  })
})
