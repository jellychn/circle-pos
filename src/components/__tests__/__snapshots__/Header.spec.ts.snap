// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header.vue > matches snapshot 1`] = `
"<div data-v-c970699f="" class="header">
  <div data-v-c970699f="" class="container">
    <icon-circle-pos-stub data-v-c970699f="" class="logo"></icon-circle-pos-stub>
  </div>
</div>"
`;
